frontend/src/Monitor/Leaderboard/Charts/GDP.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import Chart from 'Monitor/Charts/Chart';
+import LineChart from 'Monitor/Charts/LineChart';
+import {selectNested} from 'Monitor/Charts/Util';
+import {useLeaderboardContext} from '../Context';
+import {getTimeseries, avgReducer, sumReducer} from '../Selectors';
+import {registerChart} from 'Monitor/Charts/ChartRegistry';
+import {LeaderboardChartProps} from './Types';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import {makeStyles} from '@material-ui/core/styles';
+
+const Name = 'GDP';
+
+const useStyles = makeStyles((theme) => ({
+    formControl: {
+        margin: theme.spacing(1),
+        minWidth: 120,
+    },
+}));
+
+export const GDPChart: React.FC<LeaderboardChartProps> = ({data, name: n, overlay}) => {
+    const classes = useStyles();
+
+    const name = n ? n : Name;
+    const {groupState, filterState} = useLeaderboardContext();
+
+    const [mode, setMode] = React.useState<'sum' | 'avg'>('avg');
+    const onModeChange = (event: React.ChangeEvent<{value: unknown}>) => setMode(event.target.value as 'sum' | 'avg');
+
+    const reducer = mode === 'sum' ? sumReducer : avgReducer;
+    const selector = (snap: any, eid: number) => {
+        if(Math.random() > 0.95) console.log(snap)
+        return selectNested(`leaderboard/empire_summaries/${eid}/gdp/base/total_net`, snap);
```

Comment:
It might be nice to add dropdowns to allow the user to choose between the different gdp types (`net` vs `gross` and `base value` vs `market value`)

Reply:
agree, for later. I added the dropdown to VIctory and it looks bad. Here it's optional (at least of my needs), so prefer to keep it simple and beautiful.